internal/monitoring/prometheus/prometheus.go

```diff
@@ -28,6 +30,19 @@ func (m *Monitor) GetResponseTimeMetric(tags map[string]string) (monitoring.Metr
 	return m.responseTime.With(tags), nil
 }
 
+func (m *Monitor) SetAvailability(component string, value float64) error {
```

Comment:
I would stick with the pattern above used for the responseTime metric

```suggestion
func (m *Monitor) GetDependencyAvailability(tags map[string]string) (monitoring.MetricInterface, error)
```